server/cluster/cluster_worker.go

```diff
@@ -196,6 +196,16 @@ func (c *RaftCluster) HandleReportSplit(request *pdpb.ReportSplitRequest) (*pdpb
 	originRegion := proto.Clone(right).(*metapb.Region)
 	originRegion.RegionEpoch = nil
 	originRegion.StartKey = left.GetStartKey()
+
+	for _, region := range []*metapb.Region{left, right} {
+		regionInfo := core.NewRegionInfo(region, nil)
+		if err = c.HandleRegionHeartbeat(regionInfo); err != nil {
```

Comment:
It is not a real heartbeat and lacks a lot of information. I think it would be better to write an updated logic alone.

Reply:
like this: https://github.com/tikv/pd/blob/fe6fab9268d2d6fd34cd22edd1cf31a302e8dc5c/server/region_syncer/client.go#L199

Reply:
yes